frontend/src/composables/useKanban.js

```diff
@@ -0,0 +1,186 @@
+    try {
+      const res = await axios.delete(`http://localhost:3000/api/statuses/${column.id}`, {
+        data: { repo_id: column.repo_id }, withCredentials: true
+      })
+      
+      // Update UI: Move issues locally
+      const { targetStatusId } = res.data
+      const targetCol = columns.value.find(c => c.id === targetStatusId)
+      
+      if (targetCol) {
+        if (!issuesByColumn.value[targetCol.name]) issuesByColumn.value[targetCol.name] = []
+        issuesByColumn.value[targetCol.name].push(...(issuesByColumn.value[column.name] || []))
+      }
+      
+      delete issuesByColumn.value[column.name]
+      columns.value = columns.value.filter(c => c.id !== column.id)
+    } catch (err) {
+      alert(err.response?.data || "Error deleting column")
+    }
+  }
+
+  async function editColumn(columnId, newName) {
+    try {
+      await axios.put(`http://localhost:3000/api/statuses/${columnId}`, { name: newName }, { withCredentials: true })
+      if (selectedRepo.value) await selectRepo(selectedRepo.value) // Refresh to be safe or update local state
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  async function moveColumn(column, direction) {
+    try {
+      await axios.put(`http://localhost:3000/api/statuses/${column.repo_id}/${column.id}/move`, { direction }, { withCredentials: true })
+      if (selectedRepo.value) await selectRepo(selectedRepo.value)
+    } catch (err) {
+      console.error("Error moving column", err)
+    }
+  }
+
+  // --- ISSUE ACTIONS ---
+
+  async function onDragEnd(event) {
+    const { item, to } = event
+    const issueId = item.dataset.itemId
+    const newStatusName = to.closest(".card").querySelector(".card-header span").textContent.trim()
+    
+    const newStatus = columns.value.find(c => c.name === newStatusName)
+    if (!newStatus) return
+
+    try {
+      await axios.put("http://localhost:3000/api/issue-status", {
+        issue_id: parseInt(issueId),
+        repo_id: selectedRepo.value.id,
+        status_id: newStatus.id
+      }, { withCredentials: true })
+    } catch (err) {
+      console.error("Drag sync error", err)
+    }
+  }
+
+  function addIssueToBoard(newIssue) {
+    if (issuesByColumn.value[newIssue.status]) {
+      issuesByColumn.value[newIssue.status].unshift(newIssue)
+    }
+  }
+
+  async function updateIssue(issueNumber, updates) {
+    if (!selectedRepo.value) return
+    const { owner, name } = selectedRepo.value
+    
+    try {
+      const res = await axios.patch(`http://localhost:3000/api/github/issues/${owner.login}/${name}/${issueNumber}`, updates, { withCredentials: true })
+      
+      // Update local state reactivity
+      for (const colName in issuesByColumn.value) {
+        const index = issuesByColumn.value[colName].findIndex(i => i.number === issueNumber)
+        if (index !== -1) {
+          Object.assign(issuesByColumn.value[colName][index], res.data)
+          break
+        }
+      }
```

Comment:
Możecie się iterować od razu po wartościach obiektów.

Czemu `Object.assign`? Zwracacie tylko nadpisane pola? Jeżeli nie, czyli zwracacie cały obiekt, to wystarczy zwykłe nadpisanie:
```suggestion
      for (const col of Object.values(issuesByColumn.value)) {
        const index = col.findIndex(i => i.number === issueNumber)
        // Polecam: https://youtube.com/watch?v=CFRhGnuXG-4
        if (index === -1) continue
        
        col[index] = res.data
        break
      }
```

Jeżeli zwracacie tylko zaktualizowane pola, to powinno się dać uprościć jeszcze bardziej logikę:
```suggestion
      for (const col of Object.values(issuesByColumn.value)) {
        const issue = col.find(i => i.number === issueNumber)
        // Polecam: https://youtube.com/watch?v=CFRhGnuXG-4
        if (!issue) continue
        
        Object.assign(issue, res.data)
        break
      }
```


Dokumentacja:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values